### DISCUSSION TITLE
Looking for Schedule feature add.

### DISCUSSION THREAD

**Discussion by Bugman1400** [aqualinkd/AqualinkD#316] [Ideas]
Would like to see Sunrise/Sunset times added to the Schedule option. I think its typically based on what major city you live in or near. Would like to see +/- 15min increments (1 hr max) for the Sunrise/Sunset too. I think this would be handy for when you want the pool light on.

**Comment by sfeakes** [comment]
AqualinkD uses Linux Cron for scheduling, and that's not an option of Cron.  Since AqualinkD integrates to all home automation hubs, any advanced scheduling like that, I recommend using your home automation hub.  (Homeassistant / HomeKit / Domoticz) all support sunrise/sunset and are easily integrated with AqualinkD.   

- So turning pool devices on when it's raining / high winds / schedules for lights at dusk etc are best done through automation hub. 
- Standard Weekly schedules you can do either at control panel / aqualinkD or automation hub.
- Standard calendar year schedules you are best doing with AqualinkD since none of the automation hubs (or the control panel) support a whole year.

**Reply by Bugman1400** [reply]
hmm....my automation hub doesn't seem to be listed. I use HomeSeer HS4 PRO. Any plans to develop a Plugin for that one?

**Comment by sfeakes** [comment]
HomeSeer has a MQTT integration.  So you can use that since AqualinkD also has MQTT integration.  
AqualinkD has a very open and documented API, that supports WS / API / MQTT so you can make it integrate to any home automation hub.  HomeKit / Domoticz / HomeAssistant are simply the ones that I document.   But others have AqualinkD integrated into Alexa / Google / Smartthings and others.  I haven't heard of anyone using HomeSeer, but it shouldn't be a problem. Probably just a lot of reading up on how to configure HomeSeer.

**Comment by Bugman1400** [comment]
There doesn't seem to be much documentation on the MQTT. Trying to query it with MQTT Explorer and not get any results. The IP of my Raspberry PI running the AqualinkD is 192.168.1.236. I know the port is 1883. It appears the MQTT Explorer is trying to reach mqtt://192.168.1.236:1883. I assume there is no encryption, certificates, etc.....correct?

**Comment by sfeakes** [comment]
You probably need to install an MQTT broker, then configure AqualinkD to use that broker, then you will see the message.  You will also need to configure homesear to use the broker.

In the link below, scroll down a bit and it shows you how to install a MQTT broker.   You can install this on Any machine it doesn’t have to be the same machine the AqualinkD is running on.
https://github.com/sfeakes/AqualinkD/wiki#HomeKit

**Reply by Bugman1400** [reply]
hmm...I may be confused. I've had AqualinkD 2.3.2 installed on a Raspberry Pi for a while now. I assume the MQTT broker is installed on the RPi as well. I recently just upgraded to AqualinkD 2.3.7 and everything seems to work great. So, shouldn't I be able to query the MQTT broker on the RPi?

**Reply by Bugman1400** [reply]
ok..I was confused. Not sure why I thought MQTT was installed as part of the AqualinkD installation. Subsequently, I was able to install MQTT broker on the RPi Nano, re-config aqualinkd.conf as indicated in the Wiki, and viola. As you indicated, HomeSeer is able to interact with AqualinkD via MQTT. Apologies for the confusion.

**Reply by sfeakes** [reply]
Glad you got it working.

**Comment by bmwhd123** [comment]
I know this is an old thread and I'll probably try to figure out one of these sometime, but I wonder if it would be possible to take the output of your cron scheduler and manually modify the file to use heliocron? [https://github.com/mfreeborn/heliocron](url)

I realize any modifications in your UI would overwrite this file each time but maybe you could look at that sometime.

**Reply by sfeakes** [reply]
Yes, you could easily take the aqualinkd cron file and manually modify it to use heliocron.  I'd personally be more inclined to have the aqualinkd devices managed by heliocron in a separate file, and leave aqualinkd cron alone.

I used something very similar to heliocron about 20 years ago before home automation hubs were around, I was very disappoint, it was a bit of a PITA.   It works by being kicked off by cron at a particular time, then it pauses until a pre-determined amount of time, then kicks off the process.  Few issues I had with it.

- Linux doesn't like process sitting around doing nothing for extended periods of time and sometimes the pause/wait got killed. 

- If a system reboot happens in the pause time frame, it never get's kicked off. 

- Example, I live in an area where dusk times are 4~5 hours different between summer and winter, so your "pause" needs to be sitting for at least that amount of time.
ie (dusk is 4pm winter and 8pm summer, you need to schedule the dusk task to start 3:50pm). That's 4 hours of a paused task in summer.

With advancement of home automation hubs, and all the integration options AqualinkD has. These kind-a tasks are far better suited to be executed in Apple HomeKit, Home Assistant, Domoticz, HomeSeer, Alexa, Google etc. 

If want to go down this path, there are a lot of web UI projects available for cron. I'm sure it would be very straight forward to install one of those and heliocron on the PI, and have a nice graphical interface to manage the schedule.  If you've looked at aqualinkd's cron file, you can see how easy and straightforward it is to write you own commands for AqualinkD to execute.
Something like this.  https://github.com/alseambusher/crontab-ui. 

Another option would be to average the dusk / dawn (or what ever) time for each month, then set AqualinkD schedule on a monthly basis, which obviously it supports. (you could even go down to weekly or daily, but that's probably way too many entries to manage)

**Reply by bmwhd123** [reply]
Thank you. I’ll look in to how to get Alexa to do this since she manages everything else for us ;-)

**Comment by bmwhd123** [comment]
I got this working with Homebridge. Added the plugins for AqualinkD and HomebridgeToAlexa and boom, it just works. Thanks!

**Reply by sfeakes** [reply]
Great, glad it worked out for you.
